pkg/controller/plan/migration.go

```diff
@@ -773,12 +773,32 @@ func (r *Migration) execute(vm *plan.VMStatus) (err error) {
 			err = nil
 			break
 		}
-		if errs := k8svalidation.IsDNS1123Subdomain(vm.Name); len(errs) > 0 {
-			vm.NewName, err = r.kubevirt.changeVmNameDNS1123(vm.Name, r.Plan.Spec.TargetNamespace)
-			if err != nil {
-				r.Log.Error(err, "Failed to update the VM name to meet DNS1123 protocol requirements.")
+
+		// Check if user provided explicit a target virtual machine name
+		if vm.TargetName != "" {
+			vm.NewName = vm.TargetName
```

Comment:
Note: We don't need here the `k8svalidation.IsDNS1123Subdomain` because it's after input validation that we check it.